core/src/Streamly/Internal/Data/Fold/Window.hs

```diff
@@ -315,3 +318,35 @@ maximum n = fmap (fmap snd) $ range n
 {-# INLINE mean #-}
 mean :: forall m a. (Monad m, Fractional a) => Fold m (a, Maybe a) a
 mean = Fold.teeWith (/) sum length
+
+-- | Drop repeated elements within the specified tumbling window in the stream.
+-- /Pre-release/
+--
+{-# INLINE nub #-}
+nub :: (Monad m, Ord a) => Fold m (a, Maybe a) (Set.Set a)
+nub = Fold step initial extract
+
+    where
+
+    initial =
+        return $ Partial (Set.empty, Map.empty)
```

Comment:
Do not use both a Set and a Map. Just the Map should be enough. Make the signature of the function as:

```
nub :: (Monad m, Ord a) => Fold m (a, Maybe a) (Map a Int)
```

We can just return a Map of values and their refcounts. The users can use it in whatever way they want.